api/src/services/channel-info-service.ts

```diff
@@ -30,7 +30,7 @@ export class ChannelInfoService {
 		} else {
 			channelInfoPersistence = await ChannelInfoDb.searchChannelInfo(channelInfoSearch);
 		}
-		return channelInfoPersistence.map((c) => this.getChannelInfoObject(c));
+		return channelInfoPersistence?.map((channel) => this.getChannelInfoObject(channel)).filter((c) => c);
```

Comment:
Just noticed, the function is called `getChannelInfoObject` but it actually returns an array, not object?

Reply:
no it returns a single object but the map returns the array from `channelInfoPersistence `

Reply:
ah, sorry, there was another closing bracket after `this.getChannelInfoObject(channel)`